### DISCUSSION TITLE
Help in designing bracket pair colorization for Zed

### DISCUSSION THREAD

**Discussion by kiran-4444** [zed-industries/zed#11789] [Feature Requests]
I'm interested in working on #5259 . I've read the blog about how VSCode made colorization faster https://code.visualstudio.com/blogs/2021/09/29/bracket-pair-colorization

But I want to create an initial version for and optimize it later. I'm thinking to implement it as an extension first. What do you think will be a better option? 

Also, currently I'm exploring the current extension implementations to start on this and finding a hard time understanding `zed_extension_api`.  Is there a good documentation for that apart from https://github.com/zed-industries/extensions/blob/main/AUTHORING_EXTENSIONS.md?

**Comment by SomeoneToIgnore** [comment]
Oh, sorry for an overlook, found it now only but Zed now has rainbow brackets too: https://zed.dev/blog/rainbow-brackets
